src/roi/Roi.ts

```diff
@@ -328,6 +329,34 @@ export class Roi {
     };
   }
 
+  get centroid() {
+    return this.#getComputed('centroid', () => {
+      const roiMap = this.getMap();
+      const data = roiMap.data;
+      let xCoords = [];
+      let yCoords = [];
+      for (let column = 0; column < this.width; column++) {
+        for (let row = 0; row < this.height; row++) {
+          let target = this.computeIndex(row, column);
+          if (data[target] === this.id) {
+            xCoords.push(column + this.origin.column);
+            yCoords.push(row + this.origin.row);
+          }
+        }
+      }
+      let Xsum = xCoords.reduce((a, b) => {
```

Comment:
variables must start with a lowercase ! You may use here xSum

Reply:
You not nearly never use reduce. It is very slow. You can use directly use xMean from the package 'ml-spectra-processing'